### PR TITLE
feat: Add shadow DOM to duplicate-img-label check

### LINKED ISSUE #430: "duplicate-img-label" check to consider texts/images on composed tree
"duplicate-img-label" check to consider texts/images on composed tree

### PR DESCRIPTION
**Pull request by WilcoFiers**
Should close #430.

### CODE REVIEW

File: lib/checks/label/duplicate-img-label.js [hunk 1]
```diff
@@ -1,18 +1,16 @@
...
-	if (imgAlt === text &&
-	img.getAttribute('role') !== 'presentation' &&
-	axe.commons.dom.isVisible(img)) {
-		return true;
-	}
-}
+// Get all visible images in the composed tree of the current node
+const images = axe.utils.querySelectorAll(virtualNode, 'img')
+  // Ignore hidden or role=none/presentation images
+  .filter(({ actualNode }) => (axe.commons.dom.isVisible(actualNode) &&
```
Comment (WilcoFiers) [inline lib/checks/label/duplicate-img-label.js]: @dylanb Not sure if this was a bug, or if this is intentional. `isVisible` has a screenreader flag, should this function use that flag? I'm not sure why we care here if the image is positioned off screen or not. Thoughts?

File: lib/checks/label/duplicate-img-label.js [hunk 2]
```diff
@@ -1,18 +1,16 @@
...
+// Get all visible images in the composed tree of the current node
+const images = axe.utils.querySelectorAll(virtualNode, 'img')
+	// Ignore hidden or role=none/presentation images
+	.filter(({ actualNode }) => (axe.commons.dom.isVisible(actualNode) && 
+		!['none', 'presentation'].includes(actualNode.getAttribute('role'))
+	));
 
-return false;
+// See if any of the images duplicate the node's text
+return images.some(img =>
```
Comment (dylanb) [inline lib/checks/label/duplicate-img-label.js]: Shouldn't this be `!!`

Comment (WilcoFiers) [inline lib/checks/label/duplicate-img-label.js]: array.some returns a boolean.

File: test/testutils.js [hunk 3]
```diff
@@ -42,8 +42,18 @@ testUtils.checkSetup = function (content, options, target) {
 		target = options;
 		options = {};
 	}
+	// Normalize target, allow it to be the inserted node or '#target'
+	target = target || (content instanceof Node ? content : '#target');
 	testUtils.fixtureSetup(content);
-	var node = axe.utils.querySelectorAll(axe._tree[0], target || '#target')[0];
+
+	var node;
+	if (typeof target === 'string') {
```
Comment (marcysutton) [inline test/testutils.js]: We needed this testutils file after all 👍

Comment (WilcoFiers) [inline test/testutils.js]: Yeah, this is very nice to have hanging around.

**Pull request merged**